### ISSUE TITLE
Support for --manual

### ISSUE THREAD

**Issue by brianmhunt** [Daplie/letsencrypt-cli#7]
It looks like `--manual` is already in the code, but commented out.  I'm sure the problem is known, but thought it'd be a good idea if the issue were noted explicitly in an issue.  :)

Cheers

**Comment by coolaj86** [comment]
Oh, I had it there for compatibility, but I can't see a realistic use case for it so I didn't implement it.

`manual`, according to the official client, would be that it do this

```javascript
console.log("Create a file named '" + acmeChallengeKey "' in your web root containing the following string:\n" + acmeChallengeValue);
```
and would wait for you to press enter before actually issuing the challenge.

Since the `--webroot` option is now implemented, I think `--manual` is more of a backwards compatibility thing.

**Comment by brianmhunt** [comment]
Thanks @coolaj86 

I could very well be missing something, but I don't see how one could use `webroot` to validate a domain with an app deployed via e.g. Google App Engine (which has a separate `deploy` or other async process for updating the challenge-response).

With GAE one has to run the official client with `--manual`, then upload the challenge (with either an `appcfg.py update` call or some other update to notify GAE what to serve).  So it looks like this:

1. `letsencrypt` ...
2. wait for challenge/filename
3. update what AppEngine serves with `appcfg.py` or otherwise
4. press Enter on the `letsencrypt` process so it now goes and checks for the file

Point being, there's a requisite pause with `--manual` to update the server that does not appear to be present for `--webroot`.  I have been unable to find any indication of how this would work with the `webroot` challenge – hence this issue report. :)

**Comment by coolaj86** [comment]
Okay, I will put implementing this on my list of things to do.

In the meantime, how about using the official python client, which does have that implemented?

https://daplie.com/articles/lets-encrypt-on-raspberry-pi/

**Comment by brianmhunt** [comment]
Thanks @coolaj86 ; The official client works as expected.  See e.g. https://brianmhunt.github.io/articles/let-s-encrypt-acme-on-appengine/

I am not sure how the build instructions apply to this thread.

**Comment by coolaj86** [comment]
Landed in v2.1.0

* https://github.com/Daplie/letsencrypt-cli/commit/9e90c232596a0e2ea64520a9a661aa80117ad11b
* https://github.com/Daplie/le-challenge-manual

**Comment by coolaj86** [comment]
see README.md or `letsencrypt --help` for usage
